Add support for browser-sync 3

browser-sync-webpack-plugin requires browsery-sync^2 as peer-dependency https://github.com/Va1/browser-sync-webpack-plugin/blob/master/package.json#L23 

Now that browser-sync 3 was released, it generates a dependency conflict when both packages are used

<img width="1081" alt="image" src="https://github.com/Va1/browser-sync-webpack-plugin/assets/6220055/692044a6-4bf6-4a39-9bc0-f6862f88da97">

Hi, I have the same problem.

npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: browser-sync-webpack-plugin@2.3.0
npm ERR! Found: browser-sync@3.0.2
npm ERR! node_modules/browser-sync
npm ERR!   dev browser-sync@"^3.0.2" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer browser-sync@"^2" from browser-sync-webpack-plugin@2.3.0
npm ERR! node_modules/browser-sync-webpack-plugin
npm ERR!   dev browser-sync-webpack-plugin@"^2.3.0" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: browser-sync@2.29.3
npm ERR! node_modules/browser-sync
npm ERR!   peer browser-sync@"^2" from browser-sync-webpack-plugin@2.3.0
npm ERR!   node_modules/browser-sync-webpack-plugin
npm ERR!     dev browser-sync-webpack-plugin@"^2.3.0" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

--- a/package.json
+++ b/package.json
@@ -20,7 +20,7 @@
     "standard": "^11.0.1"
   },
   "peerDependencies": {
-    "browser-sync": "^2",
+    "browser-sync": "^2 || ^3",
     "webpack": "^1 || ^2 || ^3 || ^4 || ^5"
   },
   "repository": {

